### PR TITLE
feat: ✨ Added validation for load_before

### PR DESCRIPTION
**Pull request by KANAjetzt**
Checks if the same mod id is in
- `load_before` <-> `dependencies` 
- `load_before` <-> `optional_dependencies`

closes https://github.com/GodotModding/godot-mod-loader/issues/187

### CODE REVIEW

File: addons/mod_loader/classes/mod_manifest.gd [hunk 1]
```diff
@@ -91,7 +91,33 @@ func _init(manifest: Dictionary) -> void:
 	var mod_id = get_mod_id()
 	if (not is_mod_id_array_valid(mod_id, dependencies, "dependency") or
 		not is_mod_id_array_valid(mod_id, incompatibilities, "incompatibility") or
-		not is_mod_id_array_valid(mod_id, optional_dependencies, "optional_dependency")):
+		not is_mod_id_array_valid(mod_id, optional_dependencies, "optional_dependency") or
+		not validate_distinct_mod_ids_in_arrays(
```
Comment (Qubus0) [inline addons/mod_loader/classes/mod_manifest.gd:95]: might make things more readable to split this into two if statements for valid and distinct. Not much different logic wise

**Review by Qubus0 — APPROVED** [review]
getting safer by the minute. nice

File: addons/mod_loader/classes/mod_manifest.gd (hunk 1 above)
Comment (KANAjetzt) [reply addons/mod_loader/classes/mod_manifest.gd:107]: Let me know what you think - *https://github.com/GodotModding/godot-mod-loader/pull/194/commits/26065ed7601f5ba3c31bebb29fa531180b8c595c*

**Review by Qubus0 — APPROVED** [review]
yep that works

File: addons/mod_loader/classes/mod_manifest.gd [hunk 2]
```diff
@@ -249,35 +287,36 @@ static func is_semver_valid(check_version_number: String, is_silent := false) ->
...
 
 
-# Validates a mod's dependencies and incompatibilities to ensure they don't conflict.
-static func validate_dependencies_and_incompatibilities_conflicts(
+static func validate_distinct_mod_ids_in_arrays(
 	mod_id: String,
-	dependencies: PoolStringArray,
-	incompatibilities: PoolStringArray,
+	array_one: PoolStringArray,
+	array_two: PoolStringArray,
```
Comment (otDan) [inline addons/mod_loader/classes/mod_manifest.gd:293]: we probably want to have a name that is not array_one and array_two

**Pull request merged**